src/json.c

```diff
@@ -159,6 +159,70 @@ int FieldSpec_CheckJsonType(FieldType fieldType, JSONType type, QueryError *stat
 //   return ret;
 // }
 
+#define BIT_CAST(type, value) ({ \
+  __typeof__(value) value_var = (value); \
+  static_assert(sizeof(type) == sizeof(value_var), "type and value must have the same size"); \
+  const void *ptr = &(value_var); \
+  *(type *)ptr; \
+})
```

Comment:
are we ok with this (Statement Expressions) being a GCC extension and may not work with other compilers?

Reply:
Yes. The entire project is full of such usage (`arr.h` for example)